src/java.base/share/classes/java/util/StringJoiner.java

```diff
@@ -133,6 +133,34 @@ public StringJoiner(CharSequence delimiter,
         checkAddLength(0, 0);
     }
 
+    /**
+     * Constructs a {@code StringJoiner} with no characters in it using copies
+     * of the supplied {@code prefix}, {@code delimiter} and {@code suffix}.
+     * If no characters are added to the {@code StringJoiner} and methods
+     * accessing the string value of it are invoked, it will return the
+     * {@code prefix + suffix} (or properties thereof) in the result, unless
+     * {@code setEmptyValue} has first been called.
+     *
+     * @param  delimiter the sequence of characters to be used between each
+     *         element added to the {@code StringJoiner}
+     * @param  prefix the sequence of characters to be used at the beginning
+     * @param  suffix the sequence of characters to be used at the end
+     * @param  initialCapacity the number of elements that can be added before
+     *         the internal buffer needs to be resized
+     * @throws NullPointerException if {@code prefix}, {@code delimiter}, or
+     *         {@code suffix} is {@code null}
+     */
+    public StringJoiner(CharSequence delimiter,
+                        CharSequence prefix,
+                        CharSequence suffix,
+                        int initialCapacity) {
+        this(delimiter, prefix, suffix);
+        if (initialCapacity < 0) {
+            throw new IllegalArgumentException("Capacity must be non-negative");
+        }
+        elts = new String[initialCapacity];
```

Comment:
Need to be careful if `initialCapacity`  is 0. The `add` method will try to expand by doubling the existing size, which would leave it as zero-length.

Reply:
Thanks for catching this, I fixed this and added tests for the new constructor.